spec/lib/topics_bulk_action_spec.rb

```diff
@@ -160,6 +160,58 @@
     fab!(:category)
     fab!(:fist_post) { Fabricate(:post, topic: topic) }
 
+    describe "option 'perform action silently'" do
+      fab!(:watcher) { Fabricate(:user) }
+      fab!(:admin)
+
+      before do
+        Jobs.run_immediately!
+        TopicUser.change(
+          watcher,
+          topic.id,
+          notification_level: TopicUser.notification_levels[:watching],
+        )
+      end
+
+      shared_examples "perform action silently" do
+        context "when 'perform action silently' disabled" do
+          it "will send notification to users watching the topic" do
+            expect do
+              TopicsBulkAction.new(
+                admin,
+                [topic.id],
+                type: "change_category",
+                category_id: category.id,
+              ).perform!
+            end.to change { Notification.where(user: watcher).count }
+          end
+        end
+
+        context "when 'perform action silently' enabled" do
+          it "will not send notification to users watching the topic" do
+            expect do
+              TopicsBulkAction.new(
+                admin,
+                [topic.id],
+                type: "change_category",
+                category_id: category.id,
+                silent: true,
+              ).perform!
+            end.to_not change { Notification.where(user: watcher).count }
+          end
+        end
+      end
+
+      context "when 'create_revision_on_bulk_topic_moves' enabled" do
+        SiteSetting.create_revision_on_bulk_topic_moves = true
+        include_examples "perform action silently"
```

Comment:
I personally think using a shared example here makes the tests harder to read. Can you see if you can remove the shared examples?

```
it "sends the right notification when `create_revision_on_bulk_topic_moves` site setting is enabled" do
end

it "does not send a notification when `create_revision_on_bulk_topic_moves` site setting is enabled but the `silent` option is `true`" do
end
```

I think those two cases above will be sufficient?

Reply:
I doubt that there have to be 4 tests here, though they are highly similar, because this setting takes different `if` branches when turned on/off, and this spec can effectively prevent mistakes if that code is touched in the future. :thinking: